projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js

```diff
@@ -58,6 +58,13 @@ export function response( state = {}, action ) {
 				};
 				newState.results = [ ...( 'results' in state ? state.results : [] ), ...newState.results ];
 			}
+
+			// NOTE: There's a bug in the API where paginating to the next page can return an empty response.
+			//       This is most likely caused by setting excluded post types. Since `response.total` is unreliable
+			//       in this instance, manually set it to the length of the results array.
+			if ( Array.isArray( newState.results ) && newState.results.length !== newState.total ) {
+				newState.total = newState.results.length;
```

Comment:
Should we always use newState.results.length as total? As you mentioned above, response.total is not reliable for queries excluding post types. I saw it changed from 18 to 13 and then 0 following the test instructions.

Reply:
> Should we always use newState.results.length as total?

Yes; this is what's essentially being done within the conditional. If the new `results` value is not an array or if the total value already equals the new `results` length, then there's no need to update the total.